portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -60,34 +60,42 @@ public MemeRecord(String nUrl, String nComment, long nTimestamp, double nRandomI
   }
 
   private void printJsonWithOneMeme(HttpServletResponse response, PreparedQuery results) throws IOException {
+    response.setContentType("application/json;");
+
     List<Entity> memeList = new ArrayList<Entity>();
     results.asIterable().forEach(memeList::add);
     Collections.shuffle(memeList);
+    
+    memeList.stream().findFirst().ifPresent(
+        ( aMeme ) -> {
+            
+            try
+            {
+                MemeRecord aMemeRecord = new MemeRecord(
+                    "" + aMeme.getProperty("url"),
+                    "" + aMeme.getProperty("comment"),
+                    Long.parseLong( "" + aMeme.getProperty("timestamp") ),
+                    Double.parseDouble( "" + aMeme.getProperty("randomIndex") )
+                );
+
+                Gson aGson = new Gson();
+                String json = aGson.toJson( aMemeRecord );
```

Comment:
Looks like we can make it compact without losing readability:

```suggestion
                String json = newGson().toJson( aMemeRecord );
```

Or alternatively, if you want to use the same setting of Gson in every conversion, you might make a static object:
```
private static final Gson GSON = new Gson();

...
String json = GSON.toJson(...);
```